src/math/camera.rs

```diff
@@ -0,0 +1,49 @@
+use glam::{Mat4, Vec2, Vec3};
+
+pub struct Camera {
+    pub position: Vec2,
+    pub rotation: f32,
+    pub scale: Vec2,
+}
+
+impl Default for Camera {
+    fn default() -> Self {
+        Self {
+            position: Vec2{ x: 0.0, y: 0.0 },
+            rotation: 0.0,
+            scale: Vec2::ONE
+        }
+    }
+}
+
+impl Camera {
+
+    /// Gets the real size of the viewport
+    pub fn get_real_size(&mut self, viewport: Vec2) -> Vec2 {
+        Vec2 {
+            x: viewport.x/self.scale.x, 
+            y: viewport.y/self.scale.y,
+        }
+    }
+
+    /// Gets the center offset of the viewport
+    pub fn get_center_offset(&mut self, viewport: Vec2) -> Vec2 {
+        let real_size = self.get_real_size(viewport);
+        Vec2 { x: (real_size.x/2.0), y: (real_size.y/2.0) }
+    }
+
+    /// Gets the resulting matrix from the camera and viewport
+    pub fn matrix(&mut self, viewport: Vec2) -> Mat4 {
+        let real_size = self.get_real_size(viewport);
+
+        // Faster to reuse real_size, so do that instead of calling get_center_offset
+        let origin = Vec2{ x: real_size.x/2.0, y: real_size.y/2.0 };
+        let pos = Vec3{ x: self.position.x, y: self.position.y, z: -(u16::MAX as f32)/2.0};
```

Comment:
```suggestion
        let pos = self.position.extend(-(u16::MAX as f32) / 2.0);
```

Maybe even potentially this?
```suggestion
        let pos = self.position.extend(i16::MIN as f32);
```